src/runtime/smart-button-api.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2018 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {createElement} from '../utils/dom';
+import {setImportantStyles} from '../utils/style';
+import {feArgs, feUrl} from './services';
+
+/** @const {!Object<string, string>} */
+const iframeAttributes = {
+  'frameborder': '0',
+  'scrolling': 'no',
+};
+
+/**
+ * @enum {string}
+ */
+export const Theme = {
+  LIGHT: 'light',
+  DARK: 'dark',
+};
+
+
+/**
+ * The class for Smart button Api.
+ */
+export class SmartSubscriptionButtonApi {
+  /**
+   * @param {!./deps.DepsDef} deps
+   * @param {!Element} container
+   * @param {!../api/subscriptions.ButtonOptions|undefined} options
+   * @param {function()=} callback
+   */
+  constructor(deps, container, options, callback) {
+    /** @private @const {!./deps.DepsDef} */
+    this.deps_ = deps;
+
+    /** @private @const {!Window} */
+    this.win_ = deps.win();
+
+    /** @private @const {!Document} */
+    this.doc_ = this.win_.document;
+
+    /** @private @const {!web-activities/activity-ports.ActivityPorts} */
+    this.activityPorts_ = deps.activities();
+
+    /** @private @const {!HTMLIFrameElement} */
+    this.iframe_ =
+    /** @type {!HTMLIFrameElement} */ (
+        createElement(this.doc_, 'iframe', iframeAttributes));
+
+    /** @private @const {!Element} */
+    this.container_ = container;
+
+    /** @private {!../api/subscriptions.ButtonOptions|undefined} */
+    this.options_ = options;
+
+    /** @private const {?function()=} */
+    this.callback_ = callback;
```

Comment:
Should we add a callback on the button itself right away as well to handle the keyboard tab/enter case?

Reply:
Added. Though, we need a way to `focus` the `iframe->button` when user `tabs` to the outer `button`.